nearcore/src/config_validate.rs

```diff
@@ -0,0 +1,140 @@
+    config_validator.validate()
+}
+
+struct ConfigValidator<'a> {
+    config: &'a Config,
+    validation_errors: &'a mut ValidationErrors,
+}
+
+impl<'a> ConfigValidator<'a> {
+    fn new(config: &'a Config, validation_errors: &'a mut ValidationErrors) -> Self {
+        Self { config, validation_errors }
+    }
+
+    fn validate(&mut self) -> Result<(), ValidationError> {
+        self.validate_all_conditions();
+        self.result_with_full_error()
+    }
+
+    /// this function would check all conditions, and add all error messages to ConfigValidator.errors
+    fn validate_all_conditions(&mut self) {
+        if self.config.archive == false && self.config.save_trie_changes == Some(false) {
+            let error_message = format!("Configuration with archive = false and save_trie_changes = false is not supported because non-archival nodes must save trie changes in order to do do garbage collection.");
+            self.validation_errors
+                .push_errors(ValidationError::ConfigSemanticsError { error_message: error_message })
+        }
+
+        if self.config.consensus.min_block_production_delay
+            > self.config.consensus.max_block_production_delay
+        {
+            let error_message = format!(
+                "min_block_production_delay: {:?} is greater than max_block_production_delay: {:?}",
+                self.config.consensus.min_block_production_delay,
+                self.config.consensus.max_block_production_delay
+            );
+            self.validation_errors
+                .push_errors(ValidationError::ConfigSemanticsError { error_message: error_message })
+        }
+
+        if self.config.consensus.min_block_production_delay
+            > self.config.consensus.max_block_wait_delay
+        {
+            let error_message = format!(
+                "min_block_production_delay: {:?} is greater than max_block_wait_delay: {:?}",
+                self.config.consensus.min_block_production_delay,
+                self.config.consensus.max_block_wait_delay
+            );
+            self.validation_errors
+                .push_errors(ValidationError::ConfigSemanticsError { error_message: error_message })
+        }
+
+        if self.config.consensus.sync_height_threshold == 0 {
+            let error_message = format!("consensus.sync_height_threshold should not be 0");
+            self.validation_errors
+                .push_errors(ValidationError::ConfigSemanticsError { error_message: error_message })
+        }
+
+        if self.config.consensus.header_sync_expected_height_per_second == 0 {
+            let error_message =
+                format!("consensus.header_sync_expected_height_per_second should not be 0");
+            self.validation_errors
+                .push_errors(ValidationError::ConfigSemanticsError { error_message: error_message })
+        }
+
+        if self.config.consensus.min_num_peers == 0 {
```

Comment:
don't think we wan't this check. If you're running a test network with a single validator, you'd need to set this to zero. And we do that here: https://github.com/near/nearcore/blob/cd5bc7826826626e2da49f839b44af2891dfc27c/pytest/lib/cluster.py#L407